apps/lametrostop/la_metro_stop.star

```diff
@@ -0,0 +1,144 @@
+load("schema.star", "schema")
+load("secret.star", "secret")
+load("time.star", "time")
+
+API_KEY = "AV6+xWcE+yO9GVvVewmYnM4AFuY8anaOsrSB3N/P6EoMpmZH4UtrGdRw6lCZO6aFoRoWhu8bTNja0nPDV+1ZUKPME0+E6nDBxyE6xGWHm8hsAM83416wvYVcaofso+GwYs3uDxrRFE8N6D995fTTAjqtfbp7RLCInKPBQTptuOjNXd0piDP4u/fE5uNYVsV9pA=="
+DEFAULT_STOP_ID = "METRRCA:2806"
+DEFAULT_LINE_NAME = "A"
+SHORTENED_NAMES = {
+    "Union Station": "Union",
+    "APU / Citrus College": "Azuza",
+}
+
+def main(config):
+    transit_stop_id = config.get("stop_id", DEFAULT_STOP_ID)
+    transit_line = config.get("metro_line", DEFAULT_LINE_NAME)
+    api_key = secret.decrypt(API_KEY) or config.get("dev_api_key")
+    route_info = parse_api_response(get_times(transit_stop_id, api_key))
+    stop_renders = []
+    for stop_name in route_info:
+        stop_renders.append(render_stop_info(route_info, stop_name, transit_line))
+        stop_renders.append(render.Box(
+            height = 1,
+            width = 62,
+            color = "#fff",
+        ))
+    stop_renders.pop()
+    return render.Root(
+        delay = 120,
+        child = render.Box(
+            width = 64,
+            height = 32,
+            child = render.Column(
+                main_align = "start",
+                cross_align = "center",
+                children = stop_renders,
+            ),
+        ),
+    )
+
+def render_stop_info(route_info, stop_name, transit_line):
+    display_name = SHORTENED_NAMES.get(stop_name, stop_name)
+    return render.Box(
+        width = 64,
+        height = 16,
+        child = render.Row(
+            main_align = "start",
+            expanded = True,
+            cross_align = "center",
+            children = [
+                render.Box(
+                    height = 16,
+                    width = 1,
+                ),
+                render.Box(
+                    height = 11,
+                    width = 8,
+                    child = render.Text(transit_line, color = "#fff"),
+                    color = "0073bd",
+                ),
+                render.Box(
+                    height = 12,
+                    width = 2,
+                ),
+                render.Column(
+                    main_align = "start",
+                    cross_align = "start",
+                    children = [
+                        render.Marquee(
+                            width = 64,
+                            align = "start",
+                            offset_start = 5,
+                            offset_end = 8,
+                            child = render.Text(display_name),
+                        ),
+                        render.Text(get_displayed_times(route_info[stop_name], 2), font = "5x8", offset = 1, color = "FFB033"),
+                    ],
+                ),
+            ],
+        ),
+    )
+
+def get_times(stop_id, api_key):
+    # Hits AC Transit's prediction api if there are no cache hits
+    cached = cache.get(stop_id)
```

Comment:
If you haven't already, check out the http module's built in cache functionality. You can pass `ttl_seconds` straight into `http.get()` to have a request cached. Should simplify the code here a bit, and also future proof as we may start changing up the cache module a bit in the not too distant future.

Reply:
I cant seem to find the documentation or code for that http cache, is the cache key just a hash of the URL, params, headers and body of the request?? I'd be happy to implement it using that cache instead!

Reply:
I've updated the code, please let me know if I used the http cache correctly!

Reply:
Oh wow, yeah I think we need to update some docs. Thanks for the change!